imap_data_access/file_validation.py

```diff
@@ -397,3 +397,227 @@ def construct_path(self) -> Path:
+        """Generate filename from given inputs and return a AncillaryFilePath instance.
+
+        This can be used instead of the __init__ method to make a new instance:
+        ```
+        ancillary_file_path = AncillaryFilePath.generate_from_inputs("mag",
+        "mag-rotation-matrices", "20240213", "v001")
+        full_path = ancillary_file_path.construct_path()
+        ```
+
+        Parameters
+        ----------
+        description : str
+            The descriptor for the ancillary filename.
+        instrument : str
+            The instrument for the filename.
+        start_time: str, optional
+            The start time for the filename. If not provided
+            it is assumed to be valid for all times.
+        end_time: str, optional
+            The end time for the filename. If not provided,
+            the file is valid until a file with a later
+            start_date and no end_date.
+        version : str
+            The version of the data.
+        extension : str
+            The extension type of the file.
+
+        Returns
+        -------
+        str
+            The generated filename
+        """
+        time_field = start_time
+        if end_time:
+            time_field += end_time  # I think...
+        filename = (
+            f"imap_{instrument}_{description}_{time_field}_" f"{version}.{extension}"
+        )
+        return cls(filename)
+
+    def validate_filename(self) -> str:
+        """Validate the filename and populate the error message for wrong attributes.
+
+        The error message will be an empty string if the filename is valid. Otherwise,
+        all errors with the filename will be put into the error message.
+
+        Returns
+        -------
+        error_message: str
+            Error message for specific missing attribute, or "" if the file name is
+            valid.
+        """
+        error_message = ""
+
+        if any(
+            attr is None or attr == ""
+            for attr in [
+                self.mission,
+                self.instrument,
+                self.description,
+                self.version,
+                self.extension,
+            ]
+        ):
+            error_message = (
+                f"Invalid filename, missing attribute. Filename "
+                f"convention is {imap_data_access.FILENAME_CONVENTION} \n"
+            )
+        if self.mission != "imap":
+            error_message += f"Invalid mission {self.mission}. Please use imap \n"
+
+        if self.instrument not in imap_data_access.VALID_INSTRUMENTS:
+            error_message += (
+                f"Invalid instrument {self.instrument}. Please choose "
+                f"from "
+                f"{imap_data_access.VALID_INSTRUMENTS} \n"
+            )
+
+        if self.extension not in imap_data_access.VALID_FILE_EXTENSION:
+            error_message += (
+                "Invalid extension. Extension should be cdf. \n"  # TODO: Change this
```

Comment:
Yeah you should be able to utilize the `VALID_ANCILLARY_FILE_EXTENSION` variable here